### PR TITLE
Add execute streaming detectors API

### PR DESCRIPTION
**Pull request by engechas**
### Description
Adds new API for executing streaming detectors and the logic needed to handle these requests 
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by eirsep** [comment]
Can we add a concept of “post-processors” - something that works on the response of the sink call
Use case in mind: I am ingesting data from s3 as source to opensearch as sink, so I make bulk api call to ingest into opensearch. If ingestion succeeds I want to propogate the same events and the success/failure of the api to another processor (after sink) and perform some streaming rule engine process or aggregations etc.
this way we can avoid forking code logic in SAP, Alerting plugins.
Instead invest in normalizing data seen by OSI processor by adding API for mapping OCSF field names to field names in rules from SAP which OSI can leverage

**Comment by eirsep** [comment]
Can we create detectors with aggregation rules on a cluster with these changes?

File: src/main/java/org/opensearch/securityanalytics/SecurityAnalyticsPlugin.java [hunk 1]
```diff
@@ -248,7 +251,8 @@ public List<Setting<?>> getSettings() {
                 SecurityAnalyticsSettings.CORRELATION_TIME_WINDOW,
                 SecurityAnalyticsSettings.ENABLE_AUTO_CORRELATIONS,
                 SecurityAnalyticsSettings.DEFAULT_MAPPING_SCHEMA,
-                SecurityAnalyticsSettings.ENABLE_WORKFLOW_USAGE
+                SecurityAnalyticsSettings.ENABLE_WORKFLOW_USAGE,
+                SecurityAnalyticsSettings.ENABLE_STREAMING_DETECTORS
```
Comment (eirsep) [inline src/main/java/org/opensearch/securityanalytics/SecurityAnalyticsPlugin.java:255]: if i enable the setting > create detector > turn off setting will that detector run normally?

File: src/main/java/org/opensearch/securityanalytics/model/Detector.java [hunk 2]
```diff
@@ -115,13 +113,16 @@ public class Detector implements Writeable, ToXContentObject {
 
     private String findingsIndexPattern;
 
+    private Boolean streamingDetector;
```
Comment (eirsep) [inline src/main/java/org/opensearch/securityanalytics/model/Detector.java:116]: if streaming detector is true, why are we not creating workflow with `enabled=false`

**Comment by eirsep** [comment]
can you run `./gradlew integTest` on ensuring security analytics sanity is maintained and there is no regression in correctness because of the new changes

File: src/main/java/org/opensearch/securityanalytics/SecurityAnalyticsPlugin.java (hunk 1 above)
Comment (engechas) [reply src/main/java/org/opensearch/securityanalytics/SecurityAnalyticsPlugin.java:255]: No. Something has to nudge the Workflow entry in `.opendistro-alerting-config` to get the workflow marked as non-streaming.

Or the Workflow entry can be updated manually.

**Comment by eirsep** [comment]
How will pause detector work?

**Pull request closed without merging**